libs/model/src/policies/FarcasterWorker.policy.ts

```diff
@@ -110,23 +112,51 @@ export function FarcasterWorker(): Policy<typeof inputs> {
         });
         mustExist('Contest Manager', contestManager);
 
-        const contestTopic = await models.ContestTopic.findOne({
-          where: {
+        const community = await models.Community.findByPk(
+          contestManager.community_id,
+          {
+            include: [
+              {
+                model: models.ChainNode,
+                required: false,
+              },
+            ],
+          },
+        );
+        mustExist('Community with Chain Node', community?.ChainNode);
+
+        const contestManagers = [
+          {
+            url: community.ChainNode!.private_url! || community.ChainNode!.url!,
             contest_address: contestManager.contest_address,
+            actions: [],
           },
-        });
-        mustExist('Contest Topic', contestTopic);
+        ];
 
         // create onchain content from reply cast
-        const content_url = buildFarcasterContentUrl(payload.hash);
```

Comment:
same as above... policies should ideally interface with queries and commands as separate artifacts, even if only for internal purposes... to promote a potentially reusable and cleaner architecture reflected in the ES model

Reply:
will apply on next pass